File: app/src/main/java/br/com/digitalhouse/projetomarvel/repository/ComicsRepository.java
package br.com.digitalhouse.projetomarvel.repository;

import br.com.digitalhouse.projetomarvel.network.RetrofitService;
import br.com.digitalhouse.projetomarvel.pojo.ComicsResponse;
import io.reactivex.Observable;

import static br.com.digitalhouse.projetomarvel.viewmodel.ComicsViewModel.PUBLIC_KEY;

public class ComicsRepository {

    public Observable<ComicsResponse> comicsResponse() {
        return RetrofitService.Companion.getApiService().ComicsResponse("magazine", "comic", true, RetrofitService.Companion.getTimeStamp(), RetrofitService.Companion.getHash(), PUBLIC_KEY);
    }
}

File: app/src/main/java/br/com/digitalhouse/projetomarvel/repository/ComicsRepository.kt
package br.com.digitalhouse.projetomarvel.repository

import br.com.digitalhouse.projetomarvel.constants.ConstantsAPI.PUBLIC_KEY
import br.com.digitalhouse.projetomarvel.extensions.getTimeStamp
import br.com.digitalhouse.projetomarvel.network.RetrofitService.Companion.getApiService
import br.com.digitalhouse.projetomarvel.network.RetrofitService.Companion.getHash
import br.com.digitalhouse.projetomarvel.api.ComicsResponse
import io.reactivex.Observable

class ComicsRepository {
    fun comicsResponse(): Observable<ComicsResponse?>? {
        return getApiService().getAllCharacters("magazine", "comic", true, getTimeStamp(), getHash(), PUBLIC_KEY, 0)
    }
}